Fix #21576 - Function stack frame size in case of FP register in a Thumb's PUSH ##analysis

AlexanderKucherov opened the pull request:
<!--
Read https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md
* PR title must be capitalized, concise and use ##tags
* If the PR is fixing a ticket use 'Fix #1234 - ..' in the commit message
* Follow the coding style, add tests and documentation if necessary
-->

- [x] Mark this if you consider it ready to merge
- [x] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

Fix #21576

trufae commented on libr/anal/p/anal_arm_cs.c:
```diff
@@ -4049,7 +4049,7 @@ jmp $$ + 4 + ( [delta] * 2 )
 // 0x00008160    04202de5     str r2, [sp, -4]!
 // 0x000082a0    28000be5     str r0, [fp, -0x28]
-		if (REGBASE(1) == ARM_REG_FP) {
+		if ((insn->id != ARM_INS_PUSH) && (REGBASE(1) == ARM_REG_FP)) {
```
i think its cleaner to remove the fallthrough case above and duplicate the code so the if can be removed here. it will be easier to read and easier to maintain. Can you do all those changes? if not i can handle that. thanks for submiting the PR, code changes are much easier to review like this

AlexanderKucherov replied:
Ok, clear.
Should I proceed with this PR or create a new one? Sorry, I've made too many commits for such a small fix :)
And yes, please feel free to update this PR as you need it, because I'm not so familiar with the code and with the process. I would appreciate your cooperation.

trufae replied:
yep, do the change in this PR, sorry for the delay i was buried in meetings

trufae replied:
i'll merge after that change :)

AlexanderKucherov replied:
That's fine :) I've just updated it. Could you please have a look once again when you get a minute?

AlexanderKucherov commented:
Yeah, this fix affected names of local variables in one of the existing tests. I've updated the failed test and added a new one.
Thanks for the support! Now I know how to execute the tests locally before submitting PR :)

trufae commented:
Maybe you can be also interested in https://github.com/radareorg/radare2/pull/21257
